src/app_service/service/node_configuration/service.nim

```diff
@@ -239,12 +236,13 @@ proc getNimbusProxyConfig(self: Service): bool =
 proc isNimbusProxyEnabled*(self: Service): bool =
   return self.getNimbusProxyConfig()
 
-proc setNimbusProxyConfig*(self: Service, value: bool): bool =
-  var newConfiguration = self.configuration
-  newConfiguration.NimbusProxyConfig.Enabled = value
-  return self.saveConfiguration(newConfiguration)
+proc setNimbusProxyConfigEnabled*(self: Service, enabled: bool): bool =
+  # FIXME: call status-go API to update NodeConfig
+  # when this is merged https://github.com/status-im/status-go/pull/4254
+  self.configuration.NimbusProxyConfig.Enabled = enabled
+  return true
```

Comment:
do we need to always return `true` here?

Reply:
Hey @kounkou  
Vitaly wrote that he is going to merge his part soon:
https://github.com/status-im/status-go/pull/4254#issuecomment-2135113933

and here we should add a error handling similar to:
```nim
  let response = status_node_config.setLightClient(enabled)
  if not response.error.isNil:
    error "failed to set light client: ", errDescription = response.error.message
    return false
```

Reply:
Right! thanks